wgpu-core/src/device/mod.rs

```diff
@@ -730,7 +730,7 @@ impl<A: HalApi> Device<A> {
             let dimension = match desc.dimension {
                 wgt::TextureDimension::D1 => wgt::TextureViewDimension::D1,
                 wgt::TextureDimension::D2 => wgt::TextureViewDimension::D2,
-                wgt::TextureDimension::D3 => unreachable!(),
+                wgt::TextureDimension::D3 => wgt::TextureViewDimension::D3,
```

Comment:
Unfortunately, it's not that simple. The matter of fact is that 3D textures aren't really renderable in baseline WebGPU. Creating 2D views of them is not portable. See https://github.com/gpuweb/gpuweb/issues/2303 for a starting point.

In this case, we are missing a validation check that would see that the `RENDER_ATTACHMENT` usage is requested for a 3D texture and complain. This code path here is truly unreachable therefore.

Reply:
I filed https://github.com/gpuweb/gpuweb/pull/2603 to address this in the spec. We'll still need to add the relevant validation code on our side.

FYI, your workaround should be making it a 2D array texture instead of 3D.